### PR TITLE
feat(ios): UMP Consent on new arch

### PR DESCRIPTION
**Pull request by dylancom**
### Description

UMP Consent on new arch

_Note that importing types does (still) not work with codegen, so I moved the types to the spec directly._

Any folks that can test this? @TaylorDale @wjaykim @DoctorJohn @birdofpreyru

### CODE REVIEW

File: src/AdsConsent.ts [hunk 1]
```diff
@@ -75,53 +71,53 @@ export const AdsConsent: AdsConsentInterface = {
     return native.requestInfoUpdate(options);
   },
 
-  showForm(): Promise<AdsConsentInfo> {
+  showForm() {
```
Comment (mikehardy) [inline src/AdsConsent.ts:74]: interesting, so these are no longer async?
That's a breaking change for people doing old .then/.catch/etc style, but won't matter for people doing async/await (other than lint stuff)

File: src/specs/modules/NativeConsentModule.ts [hunk 2]
```diff
@@ -433,3 +309,215 @@ export interface AdsConsentUserChoices {
...
+   *     // Show a Google-rendered form
+   *     const formResult = await AdsConsent.showForm();
+   *
+   *     console.log('User consent obtained: ', formResult.status === AdsConsentStatus.OBTAINED);
+   *   }
+   * }
+   *
+   * ```
+   */
+  showForm(): Promise<AdsConsentInfo>;
```
Comment (mikehardy) [inline src/specs/modules/NativeConsentModule.ts:357]: here they are still as async though 🤔 ?

File: src/AdsConsent.ts (hunk 1 above)
Comment (mikehardy) [reply src/AdsConsent.ts:74]: I might just be misreading the type, in that now maybe typescript is inferring things and just infers the type correctly, such that it is async and is typed as described later on where I made my second comment...

**Review by mikehardy — APPROVED** [review]
Curious about the async-ness on the type signatures but can't see anything truly wrong
Testing always good though if anyone else has time

File: src/AdsConsent.ts (hunk 1 above)
Comment (dylancom) [reply src/AdsConsent.ts:74]: The type can now be inferred indeed and that's why I removed them.

**Comment by dylancom** [comment]
> Curious about the async-ness on the type signatures but can't see anything truly wrong
> 
> Testing always good though if anyone else has time

Still async. But there is the possibility now to make some methods sync for new arch. Not sure what path we should take there, introduce new methods / breaking change?

File: src/AdsConsent.ts (hunk 1 above)
Comment (dylancom) [reply src/AdsConsent.ts:74]: Types are now defined in the spec (NativeConsentModule.ts) and automatically inferred by the native layer. This is thanks to the use of the React Native Codegen tool, which bridges the gap between TypeScript types and the native implementation.

Comment (mikehardy) [reply src/AdsConsent.ts:74]: awesome, resolved on both then - thanks!

**Pull request merged**